server/controllers/authentication.coffee

```diff
@@ -9,23 +10,52 @@ localization = require '../lib/localization_manager'
 passwordKeys = require '../lib/password_keys'
 
 
+getEnv = (callback) ->
+    async.parallel [
+        (callback) ->
+            User.first (err, user) ->
+                return callback err if err
+
+                return callback null, env = username: null unless user
+
+                env = if user.public_name?.length > 0
+                    username: user.public_name
+                else
+                    username: helpers.hideEmail user.email
+                        .split ' '
+                        .map (word) -> word[0].toUpperCase() + word.slice(1)
+                        .join ' '
+                callback null, env
+
+        , (callback) ->
+            env = apps: (key for key of require('../lib/router').getRoutes())
+            callback null, env
+    ],
+    (err, results) ->
```

Comment:
True. I just wanted to keep things separated to be more flexible, but I agree, I'll rewrite it.